src/main/java/gov/nasa/jpl/mbee/mdk/generator/DocumentWriter.java

```diff
@@ -1,67 +1,29 @@
 package gov.nasa.jpl.mbee.mdk.generator;
 
-import com.nomagic.magicdraw.core.Application;
-import com.nomagic.magicdraw.core.GUILog;
-import com.nomagic.magicdraw.openapi.uml.SessionManager;
 import com.nomagic.task.ProgressStatus;
-import com.nomagic.task.RunnableWithProgress;
-import gov.nasa.jpl.mbee.mdk.docgen.docbook.DBBook;
-import gov.nasa.jpl.mbee.mdk.docgen.docbook.DBSerializeVisitor;
-import gov.nasa.jpl.mbee.mdk.model.DocBookOutputVisitor;
 import gov.nasa.jpl.mbee.mdk.model.Document;
 
-import java.io.*;
+import java.io.File;
+
 
 /**
  * runs the generation as a runnable to not stall magicdraw main thread, also
  * allow user to cancel
+ * <p>
+ * MD Model -> Document
  *
- * @author dlam
+ * @author dlam, mwilson
  */
-public class DocumentWriter implements RunnableWithProgress {
-
-    private Document dge;
-    private File realfile;
-    private File dir;
-    private boolean genNewImage;
+public class DocumentWriter extends DocumentAndPdfWriter {
```

Comment:
DocBook writing and PDF writing should be two completely separate and sequential processes. In other words DocumentWriter should only be concerned about converting the model to DocBook - not inherit from a class that does both, i.e. `DocumentAndPdfWriter`. Then a new class should be introduced, e.g. `PdfWriter` that takes a DocBook document and outputs a PDF.

Reply:
Modify DocumentWriter to extends RunnableWithProgress and made mdModel2Document method also used by DocBook2PDF as static method.